rust/crd/src/lib.rs

```diff
@@ -799,6 +829,23 @@ impl HdfsCluster {
         self.https_secret_class().is_some()
     }
 
+    pub fn has_rackawareness_enabled(&self) -> bool {
+        self.rackawareness_config().is_some()
+    }
+    pub fn rackawareness_config(&self) -> Option<String> {
+        self.spec
+            .cluster_config
+            .rack_awareness
+            .as_ref()
+            .map(|label_list| {
+                label_list
+                    .iter()
+                    .map(|label| label.to_config())
+                    .collect::<Vec<_>>()
+                    .join(";")
+            })
```

Comment:
This function should probably return `None` for

```yaml
clusterConfig: {}
```

and

```yaml
clusterConfig:
  rackAwareness: []
```

Currently it returns `Some("")` for the second case.

```suggestion
        self.spec
            .cluster_config
            .rack_awareness
            .clone()
            .filter(|label_list| !label_list.is_empty())
            .map(|label_list| {
                label_list
                    .iter()
                    .map(TopologyLabel::to_config)
                    .collect::<Vec<_>>()
                    .join(";")
            })
```